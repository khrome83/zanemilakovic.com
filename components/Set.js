// @ts-check --jsx

import PropTypes from 'prop-types';

// Convert to Grid, add Set that uses Flexbox Spacing
const Set = ({ children, gutter, desktop, tablet, mobile }) => (
  <div className={`set ${gutter && 'gutter'}`}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      .set {
        display: flex;
        justify-content: space-between;
        flex-wrap: wrap;
        padding-top: var(--outer-padding-mobile);
        margin: 0;

        @media (--tablet-viewport) {
          padding-top: var(--outer-padding-tablet);
        }

        @media (--desktop-viewport) {
          padding-top: var(--outer-padding-desktop);
        }
      }

      .set.gutter {
        margin: 0 calc(var(--outer-padding-mobile) / -2);

        @media (--tablet-viewport) {
          margin: 0 calc(var(--outer-padding-tablet) / -2);
        }

        @media (--desktop-viewport) {
          margin: 0 calc(var(--outer-padding-desktop) / -2);
        }
      }

      .set :global(.tile) {
        margin-bottom: var(--outer-padding-mobile);
        flex-basis: ${100 / mobile}%;

        @media (--tablet-viewport) {
          margin-bottom: var(--outer-padding-tablet);
          flex-basis: ${100 / tablet}%;
        }

        @media (--desktop-viewport) {
          margin-bottom: var(--outer-padding-desktop);
          flex-basis: ${100 / desktop}%;
        }
      }

      .gutter :global(.tile) {
        margin-left: calc(var(--outer-padding-mobile) / 2);
        margin-right: calc(var(--outer-padding-mobile) / 2);
        flex-basis: ${100 / (mobile + 1)}%;

        @media (--tablet-viewport) {
          margin-left: calc(var(--outer-padding-tablet) / 2);
          margin-right: calc(var(--outer-padding-tablet) / 2);
          flex-basis: ${100 / (tablet + 1)}%;
        }

        @media (--desktop-viewport) {
          margin-left: calc(var(--outer-padding-desktop) / 2);
          margin-right: calc(var(--outer-padding-desktop) / 2);
          flex-basis: ${100 / (desktop + 1)}%;
        }
      }
    `}</style>
  </div>
);

Set.defaultProps = {
  gutter: false,
  desktop: 1,
  tablet: 1,
  mobile: 1,
};

Set.propTypes = {
  children: PropTypes.node.isRequired,
  gutter: PropTypes.bool,
  desktop: PropTypes.number,
  tablet: PropTypes.number,
  mobile: PropTypes.number,
};

export { Set as default };
