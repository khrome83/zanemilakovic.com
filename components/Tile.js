// @ts-check --jsx

import PropTypes from 'prop-types';

const Tile = ({ children }) => (
  <div className="tile">
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      .tile {
        display: flex;
        flex: 1;
        padding: 1rem;
        border: 1px solid #000;
        background-color: tan;
      }
    `}</style>
  </div>
);

Tile.propTypes = {
  children: PropTypes.node.isRequired,
};

export { Tile as default };
