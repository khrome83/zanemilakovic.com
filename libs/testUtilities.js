// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
