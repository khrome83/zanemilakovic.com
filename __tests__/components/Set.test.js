import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import Set from '../../components/Set';
import Tile from '../../components/Tile';

test('Set renders correctly', () => {
  const component = renderer.create(
    <Set desktop={3} tablet={2} mobile={1}>
      <Tile>Tile 1</Tile>
      <Tile>Tile 2</Tile>
      <Tile>Tile 3</Tile>
      <Tile>Tile 4</Tile>
      <Tile>Tile 5</Tile>
      <Tile>Tile 6</Tile>
    </Set>
  );
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('Set renders correctly with Gutter', () => {
  const component = renderer.create(
    <Set desktop={4} tablet={2} mobile={1} gutter>
      <Tile>Tile 1</Tile>
      <Tile>Tile 2</Tile>
      <Tile>Tile 3</Tile>
      <Tile>Tile 4</Tile>
    </Set>
  );
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
