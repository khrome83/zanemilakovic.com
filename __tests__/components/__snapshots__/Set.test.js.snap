// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set renders correctly 1`] = `
<div
  className="set false"
>
  <div
    className="tile"
  >
    Tile 1
  </div>
  <div
    className="tile"
  >
    Tile 2
  </div>
  <div
    className="tile"
  >
    Tile 3
  </div>
  <div
    className="tile"
  >
    Tile 4
  </div>
  <div
    className="tile"
  >
    Tile 5
  </div>
  <div
    className="tile"
  >
    Tile 6
  </div>
</div>
`;

exports[`Set renders correctly with Gutter 1`] = `
<div
  className="set gutter"
>
  <div
    className="tile"
  >
    Tile 1
  </div>
  <div
    className="tile"
  >
    Tile 2
  </div>
  <div
    className="tile"
  >
    Tile 3
  </div>
  <div
    className="tile"
  >
    Tile 4
  </div>
</div>
`;
